subcommand/acl-controller/command.go

```diff
@@ -21,16 +22,23 @@ import (
 )
 
 const (
+	flagIAMRolePath           = "iam-role-path"
 	flagConsulClientSecretARN = "consul-client-secret-arn"
 	flagSecretNamePrefix      = "secret-name-prefix"
 	flagPartition             = "partition"
 	flagPartitionsEnabled     = "partitions-enabled"
 
 	consulCACertEnvVar = "CONSUL_CACERT_PEM"
+
+	// Binding rules don't support a '/' character, so we need compatible IAM role
+	// tag names for the auth method until we can address this in Consul.
+	authMethodServiceNameTag = "consul.hashicorp.com.service-name"
+	authMethodNamespaceTag   = "consul.hashicorp.com.namespace"
```

Comment:
Technically, these are new tags. We were not previously adding tags to the task IAM role. So we could use these tag names (`consul.hashicorp.com.service-name`) for the IAM role with no change to the existing tags that we add to the ECS tasks.

Reply:
Based on a quick search, it looks like validations for this are done [here](https://github.com/hashicorp/consul/blob/c786c49282d8b17115fb29aa69d9cf44a1addd0a/agent/consul/acl_endpoint.go#L860). At some point, we might want to standardize all tags on this format, but this makes sense for now.

Reply:
Yeah, off the top of my head: The binding rules use both [bexpr](https://github.com/hashicorp/go-bexpr) (in the Selector) and [HIL](https://github.com/hashicorp/hil) templates (in the BindName). bexpr supports JSON pointer syntax so I think any tag should will work there, but it's the HIL templates that support only `[a-zA-Z0-9-.]` characters in variable names.